src/config/agoric/agoric.tsx

```diff
@@ -24,10 +28,58 @@ import {
   accountBalancesQuery,
   depositParamsQuery,
   votingParamsQuery,
+  swingSetParamsQuery,
 } from "../../lib/queries.ts";
 import { selectCoinBalance } from "../../lib/selectors.ts";
 import { DepositParams, VotingParams } from "../../types/gov.ts";
 
+// Pluralize or not to pluralize.
+const s: (array: { length: number }) => string = (array) =>
+  array.length !== 1 ? "s" : "";
+
+const textEncoder = new TextEncoder();
+const bytesFormat = new Intl.NumberFormat("en", {
+  notation: "compact",
+  style: "unit",
+  unit: "byte",
+});
+const percentFormat = new Intl.NumberFormat({
+  // @ts-expect-error Until the web platform types catch up.
+  style: "percent",
+  maximimumFractionDigits: 0,
+});
+
+const chunkBundle = async (bytes: Uint8Array, chunkSizeLimit: number) => {
+  // Generate parallel arrays of the chunks and the chunk infos.
+  const sha512Bytes = await crypto.subtle.digest("sha-512", bytes);
+  const sha512 = toHex(new Uint8Array(sha512Bytes));
+  const chunks: Uint8Array[] = [];
+  const info: ChunkInfo[] = [];
+  for (let i = 0; i < bytes.byteLength; i += chunkSizeLimit) {
+    const chunk = bytes.subarray(
+      i,
+      Math.min(bytes.byteLength, i + chunkSizeLimit),
+    );
+    const sha512Bytes = await crypto.subtle.digest("sha-512", chunk);
+    const sha512 = toHex(new Uint8Array(sha512Bytes));
+    chunks.push(chunk);
+    info.push({
+      sha512,
+      sizeBytes: BigInt(chunk.byteLength),
+      state: 0,
+    });
+  }
+
+  return {
+    chunks,
+    manifest: {
+      sha512,
+      sizeBytes: BigInt(bytes.byteLength),
+      chunks: info,
+    },
+  };
+};
```

Comment:
Suggestion: avoid shadowing local bindings.
```suggestion
const getSha512Hex = async (bytes: Uint8Array) => {
  const sha512Bytes = await crypto.subtle.digest("sha-512", bytes);
  const sha512Hex = toHex(new Uint8Array(sha512Bytes));
  return sha512Hex;
};

const chunkBundle = async (bytes: Uint8Array, chunkSizeLimit: number) => {
  const bundleSha512Hex = await getSha512Hex(bytes);

  // Generate parallel arrays of the chunks and the chunk infos.
  const chunks: Uint8Array[] = [];
  const info: ChunkInfo[] = [];
  for (let i = 0; i < bytes.byteLength; i += chunkSizeLimit) {
    const chunk = bytes.subarray(
      i,
      Math.min(bytes.byteLength, i + chunkSizeLimit),
    );
    const chunkSha512Hex = await getSha512Hex(chunk);
    chunks.push(chunk);
    info.push({
      sha512: chunkSha512Hex,
      sizeBytes: BigInt(chunk.byteLength),
      state: 0,
    });
  }

  return {
    chunks,
    manifest: {
      sha512: bundleSha512Hex,
      sizeBytes: BigInt(bytes.byteLength),
      chunks: info,
    },
  };
};
```